mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -12,17 +12,23 @@ object SemVer {
       milestone: Option[Int]
   ) {
     def >(that: Version): Boolean = {
+      lazy val baseVersionEqual =
+        this.major == this.major && this.minor == that.minor && this.patch == that.patch
+
       this.major > that.major ||
       (this.major == that.major && this.minor > that.minor) ||
       (this.major == that.major && this.minor == that.minor && this.patch > that.patch) ||
       // 3.0.0-RC1 > 3.0.0-M1
-      this.releaseCandidate.isDefined && that.milestone.isDefined ||
+      baseVersionEqual && this.releaseCandidate.isDefined && that.milestone.isDefined ||
```

Comment:
So I realized the logic here was missing a bit since previously it was showing that `SemVer.isCompatibleVersion("0.10.0-M4", "0.9.10")` was true since it wasn't correctly handling the cases where there was a major/minor/patch different plus one was a milestone or release candidate. You. can see what I mean more clearly in the tests I added for this.